Update README and CHANGELOG for Passwordless (#593)
auth0/Lock.swift: README.md

## Lock Passwordless

> Lock Passwordless handles authentication using **email** and **sms** connections.

Reviewer: I think this one should be rolled back as it relates to the form style vs the Passwordless auth method. "Lock Passwordless" view as opposed to the "Lock Classic" view, allows the user to sign in with Passwordless (here you can clarify email/sms names) and Social.

See attached an example of what I mean 
![image](https://user-images.githubusercontent.com/3900123/71850062-32093e80-30b2-11ea-9bb2-f6a943132a12.png)

Author: I used the first line of [this](https://docs-content-staging-pr-8616.herokuapp.com/docs/libraries/lock-ios/v2/passwordless) docs page. But you're right, in this context is relates to the presentation style. Will roll it back.

Author: Rolled back.

> The Passwordless feature requires your application to have the *Passwordless OTP* Grant Type enabled. Check [this article](https://auth0.com/docs/clients/client-grant-types) for more information.

To use Passwordless Authentication with Lock, you need to configure it with **OIDC Conformant Mode** set to `true`.

> > It is strongly encouraged that Lock be used in OIDC Conformant mode. When this mode is enabled, it will force Lock to use Auth0's current authentication pipeline and will prevent it from reaching legacy endpoints. By default is `false`. For more information, please see the [OIDC adoption guide](https://auth0.com/docs/api-auth/tutorials/adoption).

Reviewer 2: ```suggestion
> It is strongly encouraged that Lock be used in OIDC Conformant mode. When this mode is enabled, it will force Lock to use Auth0's current authentication pipeline and will prevent it from reaching legacy endpoints. By default this is `false`. For more information, please see the [OIDC adoption guide](https://auth0.com/docs/api-auth/tutorials/adoption).
```

> > It is strongly encouraged that Lock be used in OIDC Conformant mode. When this mode is enabled, it will force Lock to use Auth0's current authentication pipeline and will prevent it from reaching legacy endpoints. By default is `false`. For more information, please see the [OIDC adoption guide](https://auth0.com/docs/api-auth/tutorials/adoption).

Reviewer: The line above says "you need to" and this one "strongly encourages" when referring to OIDC mode. Seems like it's required on one but then optional 🤔 . Let's rephrase this second paragraph to show that the OIDC mode is now required in addition to the grant type, in order for passwordless to work on this SDK. Below is just a suggestion

```suggestion
> OIDC Conformant Mode will force Lock to use Auth0's current authentication pipeline and will prevent it from reaching legacy endpoints. By default this mode is disabled. For more information, please see the [OIDC adoption guide](https://auth0.com/docs/api-auth/tutorials/adoption).
```

Author: Thanks!

> > It is strongly encouraged that Lock be used in OIDC Conformant mode. When this mode is enabled, it will force Lock to use Auth0's current authentication pipeline and will prevent it from reaching legacy endpoints. By default this is `false`. For more information, please see the [OIDC adoption guide](https://auth0.com/docs/api-auth/tutorials/adoption).

Reviewer: The line above says "you need to" and this one "strongly encourages" when referring to OIDC mode. Seems like it's required on one but then optional 🤔 . Let's rephrase this second paragraph to show that the OIDC mode is now required in addition to the grant type, in order for passwordless to work on this SDK. Below is just a suggestion

```suggestion
> OIDC Conformant Mode will force Lock to use Auth0's current authentication pipeline and will prevent it from reaching legacy endpoints. By default this mode is disabled. For more information, please see the [OIDC adoption guide](https://auth0.com/docs/api-auth/tutorials/adoption).
```
